baseapp/baseapp.go

```diff
@@ -90,7 +94,28 @@ func NewBaseApp(name string, cdc *wire.Codec, logger log.Logger, db dbm.DB, opti
 		cms:        store.NewCommitMultiStore(db),
 		router:     NewRouter(),
 		codespacer: sdk.NewCodespacer(),
-		txDecoder:  defaultTxDecoder(cdc),
+		txDecoder:  auth.DefaultTxDecoder(cdc),
+	}
+	// Register the undefined & root codespaces, which should not be used by any modules
+	app.codespacer.RegisterOrPanic(sdk.CodespaceRoot)
+	for _, option := range options {
+		option(app)
+	}
+	return app
+}
+
+// Create and name new BaseApp
+// Does not set cdc and instead takes a user-defined txDecoder.
+// TODO: Rename to NewBaseApp and remove above constructor once auth, wire dependencies removed
+func NewBaseAppNoCodec(name string, logger log.Logger, db dbm.DB, txDecoder sdk.TxDecoder, options ...func(*BaseApp)) *BaseApp {
```

Comment:
I don't think deprecating this API is worth it; adds cruft we have to clean up later. When we promise a stable `baseapp` API, we can do this, but until then I think we should just refactor all calls.

Reply:
Should we punt this to next iteration then or... ?

Reply:
No, let's go ahead and merge.